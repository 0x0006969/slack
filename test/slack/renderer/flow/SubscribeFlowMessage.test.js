const { SubscribeFlowMessage } = require('../../../../lib/slack/renderer/flow');
const repoFixture = require('../../../fixtures/repo.json');

describe('SubscribeFlowMessage rendering', () => {
  test('works for error: no-installation', async () => {
    const message = new SubscribeFlowMessage({
      error: {
        type: 'no-installation',
        url: 'https://github.com/apps/url-to-slack-app',
      },
    });
    expect(message.getRenderedMessage()).toMatchSnapshot();
  });

  test('works for error: 404', async () => {
    const message = new SubscribeFlowMessage({
      error: {
        type: '404',
      },
      input: 'atom/atom',
    });
    expect(message.getRenderedMessage()).toMatchSnapshot();
  });

  test('works for error: invalid-link', async () => {
    const message = new SubscribeFlowMessage({
      error: {
        type: 'invalid-link',
      },
      input: 'https://amazon.com/atom/atom',
    });
    expect(message.getRenderedMessage()).toMatchSnapshot();
  });

  test('works for successfully subscribed', async () => {
    const message = new SubscribeFlowMessage({
      channelId: 'C1234',
      fromRepository: repoFixture,
    });
    expect(message.getRenderedMessage()).toMatchSnapshot();
  });

  test('works for successfully unsubscribed', async () => {
    const message = new SubscribeFlowMessage({
      channelId: 'C1234',
      fromRepository: repoFixture,
      unsubscribed: true,
    });
    expect(message.getRenderedMessage()).toMatchSnapshot();
  });
  test('works for direct messages', async () => {
    const message = new SubscribeFlowMessage({
      channelId: 'D1234',
      fromRepository: repoFixture,
    });
    expect(message.getRenderedMessage()).toMatchSnapshot();
  });

  test('works for MPIM messages', async () => {
    const message = new SubscribeFlowMessage({
      channelId: 'G1234',
      fromRepository: repoFixture,
    });
    expect(message.getRenderedMessage()).toMatchSnapshot();
  });
});
