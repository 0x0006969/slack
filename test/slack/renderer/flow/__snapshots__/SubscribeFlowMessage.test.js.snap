// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeFlowMessage rendering works for MPIM messages 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "text": "Subscribed to <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`SubscribeFlowMessage rendering works for direct messages 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "text": "Subscribed to <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`SubscribeFlowMessage rendering works for error: 404 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Could not find a repository for \`atom/atom\`",
    },
  ],
  "response_type": "ephemeral",
}
`;

exports[`SubscribeFlowMessage rendering works for error: invalid-link 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "\`https://amazon.com/atom/atom\` does not appear to be a GitHub link.",
    },
  ],
  "response_type": "ephemeral",
}
`;

exports[`SubscribeFlowMessage rendering works for error: no-installation 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Looks like the app isn't installed on your repository. <https://github.com/apps/url-to-slack-app|Install the GitHub App> to proceed.",
    },
  ],
  "response_type": "ephemeral",
}
`;

exports[`SubscribeFlowMessage rendering works for successfully subscribed 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "text": "Subscribed <#C1234> to <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`SubscribeFlowMessage rendering works for successfully unsubscribed 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "text": "Unsubscribed <#C1234> from <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;
